core/src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -208,12 +208,12 @@ private Query parseQuery(String type, XContentParser parser) {
             context.setMapUnmappedFieldAsString(mapUnmappedFieldsAsString ? true : false);
             return queryParserService.parseInnerQuery(context);
         } catch (IOException e) {
-            throw new QueryParsingException(context, "Failed to parse", e);
+            throw new QueryCreationException(context, "Failed to parse", e);
```

Comment:
this will need to be refactored along the same lines as our refactoring, as mentioned in another comment